src/components/datepicker/js/datepickerDirective.js

```diff
@@ -402,9 +402,17 @@
 
     // Responds to external changes to the model value.
     self.ngModelCtrl.$formatters.push(function(value) {
+      var parsedValue = angular.isDefined(value) ? Date.parse(value) : null;
+
+      if (parsedValue) {
+        value = self.dateLocale.parseDate(parsedValue);
```

Comment:
Okay, I see. If it isn't too much trouble, since you're making a change for L#407, perhaps you can renamed the variable as `parsedNumber` to explain that a number is expected to return. I didn't realize `Date.parse` returned a Number the first time I saw it.